commands/contract/deploy.go

```diff
@@ -13,32 +13,56 @@ import (
 	"strings"
 
 	"github.com/TIBCOSoftware/dovetail-cli/config"
-	fabricc "github.com/TIBCOSoftware/dovetail-cli/hyperledger-fabric/contract"
-	"github.com/TIBCOSoftware/dovetail-cli/pkg/contract"
+	fabc "github.com/TIBCOSoftware/dovetail-cli/hyperledger-fabric/contract"
 	"github.com/spf13/cobra"
 )
 
 var (
-	id       string
-	path     string
-	version  string
-	userName string
-	orgName  string
+	id              string
+	path            string
+	policy          string
+	initArgs        string
+	channel         string
+	userName        string
+	orgName         string
+	networkConfig   string
+	networkOverride string
 )
 
 func init() {
-	//TODO: to be supported
-	//contractCmd.AddCommand(deployCmd)
+	ContractCmd.AddCommand(deployCmd)
+	ContractCmd.AddCommand(instantiateCmd)
+
 	deployCmd.Flags().StringVar(&id, "id", "", "Id of the Smart Contract")
-	deployCmd.Flags().StringVar(&path, "path", "", "Path of the Smart Contract")
-	deployCmd.Flags().StringVarP(&version, "version", "v", "", "Version of the Smart Contract")
-	deployCmd.Flags().StringVar(&userName, "user", "Admin", "User Name")
-	deployCmd.Flags().StringVar(&orgName, "org", "org1", "Organization Name")
+	deployCmd.Flags().StringVar(&path, "path", "", "Source folder of the generated Smart Contract, e.g., /path/to/hlf/src/myapp")
+	deployCmd.Flags().StringVar(&userName, "user", "Admin", "Admin user Name of the org")
+	deployCmd.Flags().StringVar(&orgName, "org", "", "Organization Name")
```

Comment:
OrgName, config and override seem very HL Fabric specific. We  have a few options to make this blockchain agnostic. Should we have our own manifest that can be blockchain specific so the commands are generic?

Reply:
it is a problem to make admin commands "blockchain agnostic" while the underneath technologies are completely different.  The "deploy" is assumed the same as "install" for hyperledger fabric, which already makes fabric users a little confused.  Besides, the "generate" command also has a couple of parameters that are corda-specific.  We probably should branch out from the very beginning, so each blockchain has its own commands that makes sense to the underneath technology. e.g., dovetail fabric generate, dovetail fabric install, and dovetail fabric instantiate, and dovetail corda generate, dovetail corda deploy, etc.